### PR TITLE
Forbid non-transforming cards to move transformed to the battlefield

### PR DESCRIPTION
**Pull request by Susucre**
#12066 Fix

### CODE REVIEW

File: Mage/src/main/java/mage/players/PlayerImpl.java [hunk 1]
```diff
@@ -4708,6 +4708,14 @@ public boolean moveCards(Set<? extends Card> cards, Zone toZone, Ability source,
                 List<ZoneChangeInfo> infoList = new ArrayList<>();
                 for (Card card : cards) {
                     fromZone = game.getState().getZone(card.getId());
+                    // 712.14a. If a spell or ability puts a transforming double-faced card onto the battlefield "transformed"
```
Comment (JayDi85) [inline Mage/src/main/java/mage/players/PlayerImpl.java:4711]: Must put that code to `maybeRemoveFromSourceZone` here:

shot_240405_194845

P.S. Looks like old code from a screenshot must be changed too

Comment (JayDi85) [reply Mage/src/main/java/mage/players/PlayerImpl.java:4711]: Code like:
if transform param then
  check transformable and is permanent
else
  check is permanent

Comment (Susucre) [reply Mage/src/main/java/mage/players/PlayerImpl.java:4711]: Uhm should check that the back face of the card is permanent if transformable, or just the card?

Comment (Susucre) [reply Mage/src/main/java/mage/players/PlayerImpl.java:4711]: That should be good (running tests locally):
image

I leave the Nightbound TODO, not sure what that one is about, but the Life // Death one should be solved.

**Pull request merged**